### PR TITLE
Adding support for table format to allow spacing in column names

### PR DESCRIPTION
**Pull request by sunishsheth2009**
## What changes are proposed in this pull request?
Adding support for table format to allow spacing in column names

## How is this patch tested?
image
If we can see there are spaces in the column names now and it doesn't fail

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Adding support to predict step for table format to allow spacing in column names

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/recipes/steps/predict.py [hunk 1]
```diff
@@ -198,6 +175,56 @@ def _run(self, output_directory):
...
+                from delta.tables import DeltaTable
+
+                DeltaTable.create().addColumns(scored_sdf.schema).property(
+                    "delta.minReaderVersion", "2"
+                ).property("delta.minWriterVersion", "5").property(
+                    "delta.columnMapping.mode", "name"
+                ).tableName(
+                    output_location
+                ).execute()
+                self.save_mode = "append"
```
Comment (jinzhang21) [inline mlflow/recipes/steps/predict.py:226]: Why is this needed given it's a new table to be created in `else` branch? `save_mode` is user input. I don't think it's a good idea to overwrite it w/o good reason.

File: mlflow/recipes/steps/predict.py [hunk 2]
```diff
@@ -198,6 +175,56 @@ def _run(self, output_directory):
...
+            else:
+                _logger.info(f"Creating a new table at {output_location}")
+                from delta.tables import DeltaTable
+
+                DeltaTable.create().addColumns(scored_sdf.schema).property(
+                    "delta.minReaderVersion", "2"
+                ).property("delta.minWriterVersion", "5").property(
+                    "delta.columnMapping.mode", "name"
+                ).tableName(
+                    output_location
```
Comment (jinzhang21) [inline mlflow/recipes/steps/predict.py:224]: Can you add some comments to indicate what this section is all about? I don't think a reader is able to understand its purpose w/o some context on Delta table.

**Review by jinzhang21 — COMMENTED** [review]
Mostly LGTM. Some minor comments. Also, mind adding a test case?

File: mlflow/recipes/steps/predict.py (hunk 1 above)
Comment (sunishsheth2009) [reply mlflow/recipes/steps/predict.py:226]: So saving to the table always checks if the table is created. 
Previous the save was implicitly creating the table if the one was not created. Now we are creating a table and saving it post that. So if the table already exists, the default save_mode doesn't work, so we need to append it. 
We are changing the save_mode only in the create case and not in the case where the table already exists. Let me know if I am missing something

let me add comments on why we are doing this in the code so it is easier for the reader to understand.

Comment (jinzhang21) [reply mlflow/recipes/steps/predict.py:226]: Got it. Makes sense to me now! Thanks!

**Review by jinzhang21 — APPROVED** [review]
